src/NavList/NavList.tsx

```diff
@@ -60,6 +87,93 @@ const Item = React.forwardRef<HTMLAnchorElement, NavListItemProps>(
   }
 )
 
+Item.displayName = 'NavList.Item'
+
+// ----------------------------------------------------------------------------
+// ItemWithSubNav (internal)
+
+type ItemWithSubNavProps = {
+  children: React.ReactNode
+  subNav: React.ReactNode
+  subNavContainsCurrentItem: boolean
+}
+
+const ItemWithSubNavContext = React.createContext<{buttonId: string; subNavId: string}>({
+  buttonId: '',
+  subNavId: ''
+})
+
+// TODO: sx prop
+// TODO: ref prop
+// TODO: Animate open/close transition
+function ItemWithSubNav({children, subNav, subNavContainsCurrentItem}: ItemWithSubNavProps) {
+  const buttonId = useSSRSafeId()
+  const subNavId = useSSRSafeId()
+  // SubNav starts open if current item is in it
+  const [isOpen, setIsOpen] = React.useState(subNavContainsCurrentItem)
+  return (
+    <ItemWithSubNavContext.Provider value={{buttonId, subNavId}}>
+      <Box as="li" aria-labelledby={buttonId} sx={{listStyle: 'none'}}>
+        <ActionList.Item
+          role="button" // Is this the best way to make ActionList.Item a button?
```

Comment:
cc @siddharthkp

Reply:
does as="button" already work with this?

Reply:
It messes with the styles: 

![CleanShot 2022-05-12 at 11 42 05](https://user-images.githubusercontent.com/4608155/168146104-5098f477-9d64-4670-a0a2-cc8d9f5a8f91.png)

I can add some button reset styles in ActionList if this is a use case you think ActionList should support.

Reply:
Done 👍